AtReconstruction/AtFitter/OpenKF/kalman_filter/TrackFitterUKF.h

```diff
@@ -0,0 +1,375 @@
+///
+/// Copyright 2022 Mohanad Youssef (Al-khwarizmi)
+///
+/// Use of this source code is governed by an GPL-3.0 - style
+/// license that can be found in the LICENSE file or at
+/// https://opensource.org/licenses/GPL-3.0
+///
+/// @author Mohanad Youssef <mohanad.magdy.hammad@gmail.com>
+/// @file unscented_kalman_filter.h
+///
+
+#ifndef UNSCENTED_KALMAN_FILTER_LIB_H
+#define UNSCENTED_KALMAN_FILTER_LIB_H
+
+#include "kalman_filter.h"
+#include "kf_util.h"
+#include "unscented_kalman_filter.h"
+
+namespace kf {
+
+/// @brief Class for fitting tracks using the Unscented Kalman Filter (UKF) algorithm.
+/// @tparam DIM_X Dimension of the state vector.
+/// @tparam DIM_Z Dimension of the measurement vector.
+/// @tparam DIM_V Dimension of the process noise vector.
+/// @tparam DIM_N Dimension of the measurement noise vector.
+template <int32_t DIM_X = 6, int32_t DIM_Z = 3, int32_t DIM_V = 2, int32_t DIM_N = 3>
+class TrackFitterUKF : public KalmanFilter<DIM_X, DIM_Z> {
```

Comment:
Name is a placeholder. Once we get a working code I think we should refactor it to make it more accessible to users, but I want to do that once we have a working version.